config/puma.rb

```diff
@@ -7,3 +7,13 @@
 rackup      DefaultRackup
 port        ENV['PORT']     || 3000
 environment ENV['RACK_ENV'] || 'development'
+
+on_worker_boot do
+  if ENV['SOFTWARE_O_O_RBTRACE']
+    # Trace objects to generate heap dump in production
+    puts "Enabling rbtrace and object allocation tracing"
+    require 'rbtrace'
+    require 'objspace'
+    ObjectSpace.trace_object_allocations_start
```

Comment:
How do you plan to get the stats out of it?

Reply:
rbtrace can connect to a running pid and dump... then you beautify the dump with [`heapy`](https://github.com/schneems/heapy) on a different machine. Something like:

```
bundle exec rbtrace -p $pid -h
```